java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogStateStore.java

```diff
@@ -52,22 +55,32 @@ private static void loadLatestSnapshots(
         LatestSnapshots latestSnapshots = new DynamoDBTransactionLogSnapshotStore(instanceProperties, tableProperties, dynamoDB).getLatestSnapshots();
         TransactionLogSnapshotSerDe snapshotSerDe = new TransactionLogSnapshotSerDe(tableProperties.getSchema(), configuration);
         try {
-            latestSnapshots.getFilesSnapshot().ifPresent(filesSnapshot -> {
+            if (latestSnapshots.getFilesSnapshot().isPresent()) {
+                TransactionLogSnapshot filesSnapshot = latestSnapshots.getFilesSnapshot().get();
+                LOGGER.info("Found latest files snapshot with last transaction number {}. Creating file reference store using this snapshot.",
+                        filesSnapshot.getTransactionNumber());
                 try {
                     builder.filesState(snapshotSerDe.loadFiles(filesSnapshot))
                             .filesTransactionNumber(filesSnapshot.getTransactionNumber());
                 } catch (IOException e) {
                     throw new UncheckedIOException(e);
                 }
-            });
-            latestSnapshots.getPartitionsSnapshot().ifPresent(partitionsSnapshot -> {
+            } else {
+                LOGGER.info("Could not find latest files snapshot. Creating empty file reference store.");
+            }
+            if (latestSnapshots.getPartitionsSnapshot().isPresent()) {
+                TransactionLogSnapshot partitionsSnapshot = latestSnapshots.getPartitionsSnapshot().get();
+                LOGGER.info("Found latest partitions snapshot with last transaction number {}. Creating partitions store using this snapshot.",
+                        partitionsSnapshot.getTransactionNumber());
                 try {
                     builder.partitionsState(snapshotSerDe.loadPartitions(partitionsSnapshot))
                             .partitionsTransactionNumber(partitionsSnapshot.getTransactionNumber());
                 } catch (IOException e) {
                     throw new UncheckedIOException(e);
                 }
-            });
+            } else {
+                LOGGER.info("Could not find latest partitions snapshot. Creating empty partitions store.");
+            }
         } catch (UncheckedIOException e) {
             throw new RuntimeException("Failed to load latest snapshots", e);
```

Comment:
This catch doesn't seems to duplicate the inner catch statements.

Reply:
Done, thanks